code/modules/karma/karma.dm

```diff
@@ -85,6 +85,9 @@ var/list/karma_spenders = list()
 		log_game("Illegal karma spending attempt detected from [key] to [M.key]. Using the same IP!")
 		to_chat(src, "<span class='warning'>You can't spend karma on someone connected from the same IP.</span>")
 		return FALSE
+	if(!M.client.wants_karma)
+		to_chat(src, "<span class='warning'>That mob has karma gains disabled.")
```

Comment:
"That player has turned off incoming karma."
Reasoning: its based on the player, not the mob, and this wording makes it clear its their explicit choice, not, say, them not being eligible.